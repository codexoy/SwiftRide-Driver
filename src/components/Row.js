import React from "react"
import { StyleSheet, View, Text } from "react-native"

export default class Row extends React.Component {
	render() {
		return (
			<View style={{ ...styles.row, ...this.props.style }}>
				<View style={styles.rightSide}>{this.props.children}</View>
				<View style={styles.leftSide}>{this.props.leftSide}</View>
			</View>
		)
	}
}

const styles = StyleSheet.create({
	row: {
		flexDirection: "row",
		width: "100%"
	},
	rightSide: {},
	leftSide: { marginLeft: "auto" }
})
