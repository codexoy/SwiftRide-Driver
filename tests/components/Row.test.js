import { Text } from "react-native"
import React from "react"

import renderer from "react-test-renderer"

import Row from "../../src/components/Row"

describe("Row", () => {
	test("view renders correctly", () => {
		const tree = renderer
			.create(
				<Row leftSide={<Text>Test</Text>} style={{ maxHeight: 30 }}>
					<Text>test</Text>
				</Row>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
