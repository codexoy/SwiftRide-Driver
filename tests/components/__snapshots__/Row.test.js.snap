// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row view renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "maxHeight": 30,
    }
  }
>
  <Text>
    test
  </Text>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 0.5,
        "marginRight": "auto",
        "marginTop": -8,
      }
    }
  >
    <Text>
      Test
    </Text>
  </View>
</View>
`;
